### PR TITLE
Follower reads tests

### PR DESCRIPTION
**Pull request by qvad**

### CODE REVIEW

File: src/configs/artifacts.conf [hunk 1]
```diff
@@ -1,7 +1,7 @@
 artifacts {
-  sample-app-tool = {
-    glob_path = "yb-sample-apps-{sample-app-tool}.jar"
-    url = "https://github.com/yugabyte/yb-stress-test/releases/download/sa_{sample-app-tool}/yb-sample-apps-{sample-app-tool}.jar"
```
Comment (qvad) [inline src/configs/artifacts.conf:4]: will fix that

File: src/apps/yugabyte/yugabyte_nemesis.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+
+class YugabyteNemesis:
+    def __init__(self,
+                 config: Config,
+                 yugabyte_app: AbstractYugabyteApp):
+        self.config: Config = config
+        self.logger = config.meta.logger
+
+        self.sudo_ssh: Connection = Connection([ConnectionConfig(host, sudo_access=True) for host in config.server_hosts], self.logger)
```
Comment (pilshchikov) [inline src/apps/yugabyte/yugabyte_nemesis.py:16]:
```suggestion
        server_hosts = []
        for host in config.server_hosts:
            host['username'] = host.get('sudo_username', host['username'])
            server_hosts.append(ConnectionConfig(host))

        self.sudo_ssh: Connection = Connection(server_hosts, self.logger)
```
And no need for `sudo_access` property

File: tools/sample-app/src/main/java/com/yugabyte/sample/common/CmdLineOpts.java [hunk 3]
```diff
@@ -199,6 +199,16 @@ public void initialize(CommandLine commandLine) throws ClassNotFoundException {
 		}
 	}
 
+    if (commandLine.hasOption("follower_reads")) {
```
Comment (pilshchikov) [inline tools/sample-app/src/main/java/com/yugabyte/sample/common/CmdLineOpts.java:202]: We should print that `follower_reads` is enabled in log

File: src/suites/follower_reads/test_sample_apps.py [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+        self.yb_nemesis = None
+
+    def suite_setup(self):
+        super().suite_setup()
+        self.yb_nemesis = YugabyteNemesis(config=self.config, yugabyte_app=self.remote_yb)
+
+    async def restart_node_async(self):
+        self.reporter.start_step("Restarting random node")
+        restarted_nodes = self.yb_nemesis.restart_random_node()
+        self.reporter.end_step(f"Node has been restarted {restarted_nodes}", True)
```
Comment (pilshchikov) [inline src/suites/follower_reads/test_sample_apps.py:48]:
```suggestion
        step_id = self.reporter.start_step("Restarting random node")
        restarted_nodes = self.yb_nemesis.restart_random_node()
        self.logger.info(f"Node has been restarted {restarted_nodes}")
        self.reporter.end_step(step_id, True)
```

File: src/suites/follower_reads/test_sample_apps.py [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+
+    def wait_steps(self, timeout_sec: int, with_restarts=False):
```
Comment (pilshchikov) [inline src/suites/follower_reads/test_sample_apps.py:50]: Needs to add some assertions on failed steps
If some workload is ended badly then failed the test
You can do that right it there or at the end of the run

File: src/suites/follower_reads/test_sample_apps.py [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+    TRUE = ["true", None]
+    FALSE = ["false", None]
+    RANDOM = ["true", "true"]
+
+    @staticmethod
+    def get_enabled_only():
+        return [FollowerReadConfig.TRUE, FollowerReadConfig.FALSE]
+
+
+class SingleWorkloadSuite(AbstractStressSuite):
```
Comment (pilshchikov) [inline src/suites/follower_reads/test_sample_apps.py:35]: Please describe how to run universe for this test, what topology, what g-flags and so on

**Review by pilshchikov — CHANGES_REQUESTED** [review]

**Review by pilshchikov — APPROVED** [review]

**Pull request merged**